google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITSystemTest.java

```diff
@@ -488,6 +488,29 @@ public void greaterThanQuery() throws Exception {
     assertEquals(2L, querySnapshot.getDocuments().get(0).get("foo"));
   }
 
+  @Test
+  public void multipleInequalityQueryOnSameProperty() throws Exception {
+    addDocument("foo", 1);
+
+    QuerySnapshot querySnapshot = randomColl
+            .whereGreaterThan("foo", 0).whereLessThanOrEqualTo("foo", 2).get().get();
+    assertEquals(1, querySnapshot.size());
+    assertEquals(1L, querySnapshot.getDocuments().get(0).get("foo"));
+  }
+
+  @Test
+  public void invalidMultipleInequalityQuery() throws Exception {
+    addDocument("foo", 1, "bar", 2);
+
+    try {
+      randomColl.whereGreaterThan("foo", 1).whereNotEqualTo("bar", 3).get().get();
+      fail();
+    } catch (Exception e) {
+      assertTrue(
+        e.getMessage().contains("INVALID_ARGUMENT: Cannot have inequality filters on multiple properties: [bar, foo]"));
```

Comment:
all the properties used in the inequality filter is listed in the error message.